ReflectionCodingExerciseWeek12/ReflectionCodingExerciseWeek12Test/UnitTest1.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Microsoft.CSharp;
+using System;
+using System.CodeDom;
+using System.Reflection;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using ReflectionCodingExerciseWeek12;
+
+namespace ReflectionCodingExerciseWeek12Test
+{
+    [TestClass]
+    public class UnitTest1
+    {
+        /// <summary>
+        /// Tests to see if an instance of Car was constructed.
+        /// </summary>
+        [TestMethod]
+        public void CarConstructionTest()
+        {
+            Car toyota = new Car("Camry", 5);
+
+            bool testTypeOf = false;
+
+            if (toyota.GetType() == typeof(Car))
+            {
+                testTypeOf = true;
+            }
+            else
+            {
+                testTypeOf = false;
+            }
+
+            Assert.IsTrue(testTypeOf);
+        }
+        /// <summary>
+        /// Tests to see if the parameters are read and write. 
+        /// If it passes, validate is equal to 2 because both properties are read and write.
+        /// </summary>
+        [TestMethod]
+        public void CarPropertyTest()
+        {
+            Car toyota = new Car("Camry", 5);
+            var carType = toyota.GetType();
+
+            int validate = toyota.GetPropertyInformation(carType);
+            bool canReadCanWrite = false;
+
+            if (validate == 2)
+            {
+                canReadCanWrite = true;
+            }
+            else
+            {
+                canReadCanWrite = false;
+            }
+
+            Assert.IsTrue(canReadCanWrite);
+        }
+        /// <summary>
+        /// Tests to see if the Drive method can be invoked.
+        /// </summary>
+        [TestMethod]
+        public void CarMethodInvokeTest()
+        {
+            Car toyota = new Car("Camry", 5);
+
+            MethodInfo driveMethod = toyota.GetType().GetMethod("Drive", BindingFlags.Public | BindingFlags.Instance);
+            driveMethod.Invoke(toyota, null);
```

Comment:
Since this is a test method, you should confirm in some way that the method was actually invoked. Perhaps the Car class could expose a Status property or something that indicates whether the car is driving after the Drive method is called.